tests/main/views/test_users.py

```diff
@@ -2107,23 +2107,52 @@ def test_published_service_count_with_different_statuses(self):
         db.session.add(BuyerEmailDomain(domain_name="bananas.org"))
         db.session.commit()
 
-        response = self.client.get('/users/check-buyer-email', query_string={'email_address': 'buyer@bananas.org'})
-        assert response.status_code == 200
-        assert json.loads(response.get_data())['valid'] is True
-
-    def test_invalid_email_is_not_ok(self):
-        response = self.client.get('/users/check-buyer-email', query_string={'email_address': 'someone@notgov.uk'})
-        assert response.status_code == 200
-        assert json.loads(response.get_data())['valid'] is False
-
-    def test_email_address_is_required(self):
-        response = self.client.get('/users/check-buyer-email')
-        assert response.status_code == 400
+    # Deprecated
+    def test_get_valid_email_is_ok_if_domain_found_in_database(self):
+        get_response = self.client.get('/users/check-buyer-email', query_string={'email_address': 'buyer@bananas.org'})
+        assert get_response.status_code == 200
+        assert json.loads(get_response.get_data())['valid'] is True
+
+    def test_post_valid_email_is_ok_if_domain_found_in_database(self):
+        post_response = self.client.post('/users/check-buyer-email',
+                                         data=json.dumps({'emailAddress': 'buyer@bananas.org'}),
+                                         content_type='application/json')
+        assert post_response.status_code == 200
+        assert json.loads(post_response.get_data())['valid'] is True
+
+    # Deprecated
+    def test_get_invalid_email_is_not_ok(self):
+        get_response = self.client.get('/users/check-buyer-email', query_string={'email_address': 'someone@notgov.uk'})
+        assert get_response.status_code == 200
+        assert json.loads(get_response.get_data())['valid'] is False
+
+        post_response = self.client.post('/users/check-buyer-email',
+                                         data=json.dumps({'emailAddress': 'someone@notgov.uk'}),
+                                         content_type='application/json')
+        assert post_response.status_code == 200
+        assert json.loads(post_response.get_data())['valid'] is False
+
+    def test_post_invalid_email_is_not_ok(self):
+        post_response = self.client.post('/users/check-buyer-email',
+                                         data=json.dumps({'emailAddress': 'someone@notgov.uk'}),
+                                         content_type='application/json')
+        assert post_response.status_code == 200
+        assert json.loads(post_response.get_data())['valid'] is False
+
+    # deprecated
+    def test_get_email_address_is_required(self):
+        get_response = self.client.get('/users/check-buyer-email')
+        assert get_response.status_code == 400
+
+    def test_post_email_address_is_required(self):
+        post_response = self.client.post('/users/check-buyer-email')
+        assert post_response.status_code == 400
```

Comment:
Ah - this test needs the `content-type` header in the request, otherwise it's always going to 400.

It'd be great if we could also assert the error message as well here, as it will highlight the eventual change to the contract that we'll be introducing with the `json_has_required_keys` function (some of our frontend apps rely on the content of the error message returned by the API, so it's good to know what that'll be). 

With the content type included, for the test above we can assert the value of  `json.loads(post_response.get_data())['error']`.

Reply:
Great suggestion @katstevens and this also flushed out issues with the code and the tests. Have updated now in https://github.com/alphagov/digitalmarketplace-api/pull/1037/commits/0bf90778add324e0048d674ab15cf179d5e2caa9.